content/docs/components-and-props.md

```diff
@@ -16,23 +16,24 @@ prev: rendering-elements.html
 next: state-and-lifecycle.html
 ---
 
-Components let you split the UI into independent, reusable pieces, and think about each piece in isolation. This page provides an introduction to the idea of components. You can find a [detailed component API reference here](/docs/react-component.html).
+componentहरूबाट तपाईंले UI लाई स्वतन्त्रा रूपमा  विभाजन  गर्न , पुन: प्रयोज्य टुक्राहरूमा विभाजन गर्न  र प्रत्येक टुक्रालाई अलग अलग  रूपमा सोच्न सक्नुहुन्छ।यस पृष्ठले componentsको विचारलाई  परिचय प्रदान गर्दछ। तपाईंले यो linkमा क्लिक गर्नुभयो भने  componet APIको बारेमा विस्तृत  जानकारी प्राप्त गर्न सक्नुहुन्छ [detailed component API reference here](/docs/react-component.html).
 
-Conceptually, components are like JavaScript functions. They accept arbitrary inputs (called "props") and return React elements describing what should appear on the screen.
+अवधारणागत रूपमा, componentहरू JavaScript functions जस्तै हुन्। तिनीहरू मध्यस्थ इनपुट  (“props” भनिन्छ) लिन्छन् र स्क्रिनमा के देखा पर्दछ भनेर वर्णन गर्ने Reactको elements फिर्ता गर्दछन्।
 
-## Function and Class Components {#function-and-class-components}
 
-The simplest way to define a component is to write a JavaScript function:
+## Function र Class Components {#function-and-class-components}
+
+component परिभाषित गर्ने सब भन्दा साधारण तरीका भनेको JavaScript functions लेख्नु हो:
 
 ```js
 function Welcome(props) {
   return <h1>Hello, {props.name}</h1>;
 }
 ```
 
-This function is a valid React component because it accepts a single "props" (which stands for properties) object argument with data and returns a React element. We call such components "function components" because they are literally JavaScript functions.
+यो function मान्य React component हो किनकि यसले एकल “props” ("proporties" भनिन्छ) object आर्गुमेन्ट डाटाको साथ लीने गर्दछ र React एलिमेट्स फिर्ता गर्दछ।  त्यस्ता कम्पोनेन्टहरूलाई हामी  “function components”  भन्छौं किनभने तिनीहरू शाब्दिक JavaScript functions हुन्। 
```

Comment:
Since props is a shortened form for properties, would we better describe the intent by saying `(जुन "properties" को छोटकरी हो)` instead of `("proporties" भनिन्छ)` ? 

For consistency, let's not transliterate elements? Change `एलिमेट्स` to `elements`
Similary, since "components" are not transliterated: Should we change  `कम्पोनेन्टहरूलाई` to `component हरूलाई` ?